sootup.analysis/src/main/java/sootup/analysis/interprocedural/icfg/JimpleBasedInterproceduralCFG.java

```diff
@@ -148,6 +150,21 @@ public JimpleBasedInterproceduralCFG(
     initializeStmtToOwner();
   }
 
+  public String buildICFGGraph() {
+    ArrayList<StmtGraph> stmtGraphSet = new ArrayList<>();
+    // To Sort the methodSignature set with entrypoint as the first element.
+    Set<MethodSignature> methodSignatures = cg.getMethodSignatures();
+    methodSignatures.remove(mainMethodSignature);
+    Set<MethodSignature> modifiedMethodSignatures = new LinkedHashSet<>();
```

Comment:
We don't need to get all the method signatures upfront. you can start from the main method and traverse the call graph using `cg.getCallsFrom()`. You can then add stmtGraphs to the dot graph as you traverse and add corresponding edges that represent the calls.

Reply:
CallGraph is not accessible in ICFGDotExporter.java, I need to add a dependency for accessing callgraph. So this is the plausible way of doing it I reckon. I can't use cg.callsFrom() because I need the hashcode of the invoke statement.

Reply:
What do you mean it's not accessible? You are calling `cg.getMethodSignatures()` ?